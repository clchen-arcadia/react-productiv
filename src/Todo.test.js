import { render, fireEvent } from "@testing-library/react";
import Todo from "./Todo";

it("renders without crashing", function () {
  const { container } = render(
    <Todo
      todo={{ id: 1, title: "TEST", description: "Testing", priority: 1 }}
    />
  );
});

it("contains expected term", function () {
  const result = render(
    <Todo
      todo={{ id: 1, title: "TEST", description: "Testing", priority: 1 }}
    />
  );
  expect(result.queryByText("TEST")).toBeInTheDocument();
  expect(result.queryByText("Testing")).toBeInTheDocument();
});

it("matches snapshot", function () {
  const { container } = render(
    <Todo
      todo={{ id: 1, title: "TEST", description: "Testing", priority: 1 }}
    />
  );
  expect(container).toMatchSnapshot();
});
