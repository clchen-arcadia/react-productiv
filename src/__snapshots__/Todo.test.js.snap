// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        TEST
      </b>
       
      <small>
        (priority: 
        1
        )
      </small>
    </div>
    <div>
      <small>
        Testing
      </small>
    </div>
  </div>
</div>
`;
